packages/node/src/policy-gen/policy-gen-compartment-class.js

```diff
@@ -285,7 +302,9 @@ export const makePolicyGenCompartment = (compartmentMap, policyOverride) => {
           !isRecordModuleDescriptor(moduleDescriptor) &&
           !isSourceModuleDescriptor(moduleDescriptor)
         ) {
-          // unknown if this ever happens
+          // This would mean a new module descriptor type was added in
+          // @endo/compartment-mapper.
+          // TODO: consider throwing an error here
```

Comment:
If we don't throw here, we might miss when a new Module Descriptor shape is added.

Reply:
I'm wishy-washy on this, but guess it couldn't hurt.

- [ ] throw here
- [ ] add a mutation-like test where we receive a malformed `ModuleDescriptor` in a `CompartmentDescriptor`